go/vt/vttablet/tabletserver/tx_engine.go

```diff
@@ -411,58 +411,129 @@ func (te *TxEngine) shutdownLocked() {
 // to ensure there are no future collisions.
 func (te *TxEngine) prepareFromRedo() error {
 	ctx := tabletenv.LocalContext()
-	var allErr concurrency.AllErrorRecorder
-	prepared, failed, err := te.twoPC.ReadAllRedo(ctx)
-	if err != nil {
-		return err
+
+	prepared, failed, readErr := te.twoPC.ReadAllRedo(ctx)
+	if readErr != nil {
+		return readErr
 	}
 
-	maxid := int64(0)
-outer:
-	for _, preparedTx := range prepared {
-		txid, err := dtids.TransactionID(preparedTx.Dtid)
+	var (
+		maxID           = int64(0)
+		preparedCounter = 0
+		failedCounter   = len(failed)
+		lastDtid        string
+		lastErr         error
+		allErrs         []error
+	)
+
+	checkErr := func(dtid string, err error) {
 		if err != nil {
-			log.Errorf("Error extracting transaction ID from dtid: %v", err)
+			allErrs = append(allErrs, vterrors.Wrapf(err, "dtid - %v", dtid))
+			if te.checkErrorAndMarkFailed(ctx, dtid, err, "TwopcPrepareRedo") {
+				failedCounter++
+			}
 		}
-		if txid > maxid {
-			maxid = txid
+	}
+
+outer:
+	for _, preparedTx := range prepared {
+		var conn *StatefulConnection
+
+		txID, _ := dtids.TransactionID(preparedTx.Dtid)
```

Comment:
I don't get it - why are we ignoring the error here?

Reply:
After talking it over with H - maybe we could solve this by extracting the inside of the loop into a method, and using a defer in that method to check errors